floodsub.go

```diff
@@ -126,12 +132,16 @@ func (p *PubSub) processLoop(ctx context.Context) {
 			}
 		case treq := <-p.getTopics:
 			var out []string
-			for t := range p.myTopics {
-				out = append(out, t)
+			for t, subs := range p.myTopics {
+				if len(subs) > 0 {
```

Comment:
should this ever not be > 0? That seems like it could end up being a memory leak